From: Dan Williams <dan.j.williams@intel.com>
Date: Thu, 16 Jun 2011 17:20:35 -0700
Subject: [PATCH 249/273] isci: fix smp response frame overrun
Git-commit: 5edc33480c1c363ab361a881f2957b9fba5185cf
References: FATE#311808,bnc#709528
Patch-Mainline: 3.0

Due to a typo we currently copy way too much when copying over the
response data, but since a request is likely backed by a full page
allocation we don't corrupt live data.

Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/isci/request.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/scsi/isci/request.c b/drivers/scsi/isci/request.c
index ebe160c..f4fbca7 100644
--- a/drivers/scsi/isci/request.c
+++ b/drivers/scsi/isci/request.c
@@ -1694,7 +1694,7 @@ scic_sds_io_request_frame_handler(struct scic_sds_request *sci_req,
 								      frame_index,
 								      &smp_resp);
 
-			word_cnt = (sizeof(struct smp_req) - SMP_RESP_HDR_SZ) /
+			word_cnt = (sizeof(struct smp_resp) - SMP_RESP_HDR_SZ) /
 				sizeof(u32);
 
 			sci_swab32_cpy(((u8 *) rsp_hdr) + SMP_RESP_HDR_SZ,
-- 
1.6.0.2

